python/cuspatial/cuspatial/utils/join_utils.py

```diff
@@ -31,7 +32,7 @@ def apply_binarize(in_col, width):
     if out.size > 0:
         out[:] = 0
         binarize.forall(out.size)(in_col, out, width)
-    return out
+    return cp.asarray(out)
```

Comment:
If this is 2D, you could use `asfortranarray` so when sliced into 1D columns the memory is contiguous. (I'll also ensure this on the cudf side)

Reply:
`asfortranarray` will reorder the data to F-contiguous with new allocation and increases peak memory usage. It would be nice if we don't have to assume memory order for return values? Cupy support fast indexing for any memory order anyway.

Reply:
If we would like to change the `order` of the array, this could be done by passing [the `order` flag to `reshape`]( https://docs.cupy.dev/en/stable/reference/generated/cupy.reshape.html )

That said, given this wasn't done before with the `DeviceNDArray` (and may involve other subtle changes in associated code), maybe this should be converted to a new issue/PR?

Reply:
> maybe this should be converted to a new issue/PR?

Yes - again, unless explicitly required by downstream libraries, I would rather cuspatial not having to dictate a memory order here but just stick to the defaults. Switching to cupy gives the flexibility not having to be explicit on memory orders, and [cudf is adding back support to](https://github.com/rapidsai/cudf/pull/16436) different memory order arrays. This relieves the responsibility from cuspatial.